api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -243,6 +248,19 @@ public DbWorkspace getRequiredWithCohorts(String ns, String firecloudName) {
     return workspace;
   }
 
+  @Override
+  public void requireActiveBilling(String workspaceNamespace, String workspaceId)
+      throws ForbiddenException {
+    if (!workbenchConfigProvider.get().featureFlags.enableBillingLockout) {
+      return;
+    }
+
+    if (BillingStatus.INACTIVE.equals(
+        getRequired(workspaceNamespace, workspaceId).getBillingStatus())) {
+      throw new ForbiddenException("Workspace is in an inactive billing state");
```

Comment:
this is potentially ambiguous, should this include the workspace ID in the message? Alternatively this method could return bool and the controller could throw a contextually relevant message.